homeassistant/components/weather/__init__.py

```diff
@@ -114,52 +190,125 @@ class WeatherEntity(Entity):
     _attr_humidity: float | None = None
     _attr_ozone: float | None = None
     _attr_precision: float
-    _attr_pressure: float | None = None
+    _attr_native_pressure: float | None = None
     _attr_pressure_unit: str | None = None
+    _attr_native_pressure_unit: str | None = None
     _attr_state: None = None
-    _attr_temperature_unit: str
-    _attr_temperature: float | None
-    _attr_visibility: float | None = None
+    _attr_temperature_unit: str | None = None
+    _attr_native_temperature_unit: str
+    _attr_native_temperature: float
+    _attr_native_visibility: float | None = None
     _attr_visibility_unit: str | None = None
+    _attr_native_visibility_unit: str | None = None
+    _attr_native_precipitation: float | None = None
     _attr_precipitation_unit: str | None = None
+    _attr_native_precipitation_unit: str | None = None
     _attr_wind_bearing: float | str | None = None
-    _attr_wind_speed: float | None = None
+    _attr_native_wind_speed: float | None = None
     _attr_wind_speed_unit: str | None = None
+    _attr_native_wind_speed_unit: str | None = None
+
+    _weather_option_temperature_uom: str | None = None
+    _weather_option_pressure_uom: str | None = None
+    _weather_option_visibility_uom: str | None = None
+    _weather_option_precipitation_uom: str | None = None
+    _weather_option_wind_speed_uom: str | None = None
+
+    _deprecated_weather_entity_report = False
+    _deprecated_weather_entity_reported = False
+
+    def __init_subclass__(cls, **kwargs: Any) -> None:
+        """Post initialisation processing."""
+        super().__init_subclass__(**kwargs)
+        if "temperature_unit" in cls.__dict__:
+            cls._deprecated_weather_entity_report = True
```

Comment:
Maybe just report it directly here, something similar is done here: https://github.com/home-assistant/core/pull/73552
Also, to facilitate backwards compatibility, I think it should be possible to do:
```py
cls.__dict__["_temperature_unit_override"] = cls.__dict__["temperature_unit"]
cls.__dict__["temperature_unit"] = WeatherEntity__dict__["temperature_unit"]
```

Reply:
Tried both this and using setattr but both gives `TypeError: can't apply this __setattr__ to ABCMeta object`

Reply:
Try this:
```py
setattr(cls, "_temperature_unit_override"], cls.__dict__["temperature_unit"])
setattr(cls, "temperature_unit", WeatherEntity__dict__["temperature_unit"])
```